velox/functions/sparksql/MapFromArrays.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <common/base/Exceptions.h>
+#include <core/QueryConfig.h>
+#include <vector/ComplexVector.h>
+#include <vector/DecodedVector.h>
+#include <vector/TypeAliases.h>
+#include <unordered_set>
+
+#include "velox/expression/EvalCtx.h"
+#include "velox/expression/Expr.h"
+#include "velox/expression/VectorFunction.h"
+
+// Returns a map created using the given key/value arrays.
+// See documentation at
+// https://spark.apache.org/docs/latest/api/sql/#map_from_arrays
+//
+// Example:
+// Select map_from_arrays(array(1,2,3), array('a','b','c'));
+//
+// Result:
+// {1:"a",2:"b",3:"c"}
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+template <bool throwExceptionOnDuplicateMapKeys>
+class MapFromArraysFunction : public exec::VectorFunction {
```

Comment:
I suppose we can reuse the implementation of `MapFunction` by registering it as a stateful vector function to respect  the config `throw_exception_on_duplicate_map_keys` in sparksql. For Presto, we keep registering it with `allowDuplicateKeys` being false. How do you think?

Reply:
the behavior of `MapFunction` if different with Spark, in that when there are duplicate keys, it doesn't **deduplate**.
for example, 
`map_from_array(array(1,1), array('a', 'b'))`, 
returns 
`map(1->a, 1->b)` in Presto/MapFunction,
but returns
`map(1->b)` in Spark

Reply:
Can we do some simple refactor for `MapFunction`, looks like it is only different with throw_exception_on_duplicate_map_keys config.

Reply:
ok, will try to refactor it